### PR TITLE
doc: add initial doc for consumers

### PR DESCRIPTION
**Pull request by mhdawson**
Signed-off-by: Michael Dawson <michael_dawson@ca.ibm.com>

### CODE REVIEW

File: doc/command-line-usage.md [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+* Package Maintainers - document and communicate expectations to consumers
+* Package Consumers - understand more about their key dependencies and where they 
+  should invest.
+
+The sections which follow document the usage/options of interest for each use
+case.
+
+## Package Consumers
+
+The support tool helpss package conumers review and understand the
```
Comment (rodion-arr) [inline doc/command-line-usage.md]:
```suggestion
The support tool helps package consumers review and understand the
```

File: doc/command-line-usage.md [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+package support information provided by maintainers. See
+[Package-support.md](https://github.com/nodejs/package-maintenance/blob/master/docs/PACKAGE-SUPPORT.md)
+which documents the suggested best practice and the spedific format of the
```
Comment (rodion-arr) [inline doc/command-line-usage.md]:
```suggestion
which documents the suggested best practice and the specific format of the
```

File: doc/command-line-usage.md [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+package support information.
+
+Package consumers use the `show` command to display the support information
+provided by package maintainers.
+
+It is understood that this is only the most basic support that can be provided
+to the consumer but the current focus for the support tool is to provide features
+to encourage/support maintainers to add the package support info to their packages.
+In the future more addvanced options can be added to the show command to provide
```
Comment (rodion-arr) [inline doc/command-line-usage.md]:
```suggestion
In the future more advanced options can be added to the show command to provide 
```

File: doc/command-line-usage.md [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+Package consumers use the `show` command to display the support information
+provided by package maintainers.
+
+It is understood that this is only the most basic support that can be provided
+to the consumer but the current focus for the support tool is to provide features
+to encourage/support maintainers to add the package support info to their packages.
+In the future more addvanced options can be added to the show command to provide 
+more deduplicated or focussed views of the information available.
+
+The general flow ia:s
```
Comment (rodion-arr) [inline doc/command-line-usage.md]:
```suggestion
The general flow is: 
```

File: doc/command-line-usage.md [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+In the future more addvanced options can be added to the show command to provide 
+more deduplicated or focussed views of the information available.
+
+The general flow ia:s 
+
+* install node.js (version 10.x or later)
+* install the package support tool
+* install the package for which they would like to display information
+* cd into the directory which contains the root package
+* execute `suport show`
```
Comment (rodion-arr) [inline doc/command-line-usage.md]:
```suggestion
* execute `support show`
```

File: doc/command-line-usage.md [hunk 6]
```diff
@@ -0,0 +1,227 @@
...
+In this case it also ONLY shows support info for the `support` package as the support information has not
+been added to packages yet.
+
+The support info shown is:
+
+```shell
+  @pkgjs/support(0.0.2) - { "versions": [ { "version": "*", "target": { "node": "supported" }, "response": { "type": "best-effort" }, "backing": { "hobby": "https://github.com/pkgjs/support" } } ] }
+```
+
+For the rest of the modules it simply shows `unknown` as the the packages do not yet provide it.
```
Comment (rodion-arr) [inline doc/command-line-usage.md]:
```suggestion
For the rest of the modules it simply shows `unknown` as the packages do not yet provide it.
```

File: doc/command-line-usage.md [hunk 7]
```diff
@@ -0,0 +1,227 @@
...
+
+The package support tool prefers support information which is available locally with the package after installation.
+This is to allow offline use. The documented best practices for adding support info are written to maximmize
```
Comment (rodion-arr) [inline doc/command-line-usage.md]:
```suggestion
This is to allow offline use. The documented best practices for adding support info are written to maximize
```

File: doc/command-line-usage.md [hunk 8]
```diff
@@ -0,0 +1,227 @@
...
+```shell
+support  --base-path=${cwd}/../.. show
+```
+
+In the case where the required base path is not specified and the file is outside
+of the package the tool will simply print the url form which the support information
+can be obtained versus displaying it. For example:
+
+```
+@pkgjs/support-show-local-escape-ok(0.0.1) - https://github.com/pkgjs/support/blob/master/my-support-info.jso
```
Comment (rodion-arr) [inline doc/command-line-usage.md]:
```suggestion
@pkgjs/support-show-local-escape-ok(0.0.1) - https://github.com/pkgjs/support/blob/master/my-support-info.json
```

File: doc/command-line-usage.md [hunk 9]
```diff
@@ -0,0 +1,227 @@
...
+
+Package maintainers want to add and manage the support information for their modules.
+
+Ultimately we'll want commands help generate the support info, but as a first step
+we are provided the `validate` command which validates that the info is in the
+right format/locations. 
+
+To add support information a maintainer needs to:
+
+* add information to the package.json which specifices were to find the support info
```
Comment (rodion-arr) [inline doc/command-line-usage.md]:
```suggestion
* add information to the package.json which specifies were to find the support info
```

File: doc/command-line-usage.md [hunk 10]
```diff
@@ -0,0 +1,227 @@
...
+* provide a file with the support info.
+
+The `validate` command checks that both the information added to the package.json
+and the the file containing the support info (package-support.json by default) is valid.
```
Comment (rodion-arr) [inline doc/command-line-usage.md]:
```suggestion
and the file containing the support info (package-support.json by default) is valid.
```

File: doc/command-line-usage.md [hunk 11]
```diff
@@ -0,0 +1,227 @@
...
+`validate` is run from the directory that contains the package.json for the package.
+Depending the how the `support` section in the package.json is configured validate will:
+
+* validate the format of the `support` section in the package.json conforms to the
+documentation in [Integration into package.json](https://github.com/nodejs/package-maintenance/blob/master/docs/PACKAGE-SUPPORT.md#integration-into-packagejson)
+* Calculate the file which cotains the support info
+* If the file is available locally, use that data unless the `--canonical`
+  option was specified.
+* If the file is not available locally or `--canonical` was specified
+  * If --fetch was provided as an option, fetch the file and use thate data
```
Comment (rodion-arr) [inline doc/command-line-usage.md]:
```suggestion
  * If --fetch was provided as an option, fetch the file and use that data
```

File: doc/command-line-usage.md [hunk 12]
```diff
@@ -0,0 +1,227 @@
...
+
+If the information support information is not avialable locally or you have specified
+canonical without `--fetch` you'll see something like:
+
+```shell
+support info not resolved: https://github.com/pkgjs/support/blob/master/package-support.json
+```
+
+in this case the file is local to the package, but --canonical was specified. The same
+would be shown if the info in the package.json file pointed to a file in a different pacakge.
```
Comment (rodion-arr) [inline doc/command-line-usage.md]:
```suggestion
would be shown if the info in the package.json file pointed to a file in a different package.
```

**Comment by mhdawson** [comment]
@rodion-arr thanks for reviewing the doc, applied your suggestions.

**Comment by mhdawson** [comment]
@rodion-arr can you add your approval to the PR, need that so I can go ahead and land.

**Review by rodion-arr — APPROVED** [review]

**Comment by mhdawson** [comment]
@wesleytodd, @ljharb do you have time to take a look before I land this?

File: README.md [hunk 13]
```diff
@@ -23,28 +23,24 @@ The spec proposal is currently being reviewed and is open for feedback.  As we h
...
-  // The error is annotated with the
-  // errors and schema from `ajv`
-  console.error(e)
-  console.log(e.validationErrors)
-  console.log(e.validationSchema)
-}
-```
+## Command line usage
+
+A command line tool is provided which supports the following options:
```
Comment (ljharb) [inline README.md]:
```suggestion
A command line tool is provided which supports the following commands:
```

File: doc/command-line-usage.md [hunk 14]
```diff
@@ -0,0 +1,227 @@
...
+to the consumer. The current focus for the support tool is to provide features
+to encourage/support maintainers to add the package support info to their packages.
+In the future more advanced options can be added to the show command to provide 
+more deduplicated or focussed views of the information available.
+
+The general flow is:
+
+* install node.js (version 10.x or later) and ensure it is on the path
+* npm install the package support tool global (or optionally locally in which
+  case the `bin` directory need to be add the current path)
```
Comment (ljharb) [inline doc/command-line-usage.md]: why not `npx` instead of globally installing?

File: doc/command-line-usage.md [hunk 15]
```diff
@@ -0,0 +1,227 @@
...
+For the rest of the modules it simply shows `unknown` as the packages do not yet provide it.
+
+The package support tool prefers support information which is available locally within the package after installation.
+This is to allow offline use. The documented best practices for adding support info are written to maximize
+the likelyhood that at reasonably up to date copy of the support information is provided as part of the npm
+install.
+
+If the consumer wishes to ignore the local information and always use the canonical version of the support information
+they can add the `--canonical` option to the command. Instead of using the local support info, if available, it will
+display the URL from which the canonical information can be retrieved. This is the same behaviour when
```
Comment (ljharb) [inline doc/command-line-usage.md]:
```suggestion
display the URL from which the canonical information can be retrieved. This is the same behavior when 
```
unless we have a standard of using british spelling?

**Review by ljharb — APPROVED** [review]

File: doc/command-line-usage.md (hunk 15 above)
Comment (mhdawson) [reply doc/command-line-usage.md]: No that's just the Canadian in me.  The standard is US english :)

File: doc/command-line-usage.md (hunk 14 above)
Comment (mhdawson) [reply doc/command-line-usage.md]: good point, I'll try that out and update

File: doc/command-line-usage.md [hunk 16]
```diff
@@ -0,0 +1,230 @@
...
+
+It is understood that this is only the most basic support that can be provided
+to the consumer. The current focus for the support tool is to provide features
+to encourage/support maintainers to add the package support info to their packages.
+In the future more advanced options can be added to the show command to provide 
+more deduplicated or focussed views of the information available.
+
+The general flow is:
+
+* install the package to display package support information
```
Comment (mhdawson) [inline doc/command-line-usage.md]:
```suggestion
* install the package to display package support information for
```

File: doc/command-line-usage.md (hunk 14 above)
Comment (mhdawson) [reply doc/command-line-usage.md]: @ljharb updated

**Review by ljharb — APPROVED** [review]

**Pull request merged**